ec/src/hashing/curve_maps/swu/mod.rs

```diff
@@ -31,6 +28,10 @@ pub struct SWUMap<P: SWUParams> {
     curve_params: PhantomData<fn() -> P>,
 }
 
+fn parity<F: Field>(element: &F) -> bool {
```

Comment:
Potentially it makes sense to add this under the `SWUParams` trait, since this is the only context one would use it. Or even better: create a separate trait in this module (e.g. `FieldParity`) and just implement it here:
```
impl Field Parity for SWUMap {
   ...
}
```
This way, if anyone ever wants to use the `parity()` method, they can just import the `Parity` trait from this module, without needing to define `XI` etc.

Thoughts?

Reply:
@mmagician, you are right, there are multiple possibilities of adding parity to swu/mod.rs; the one I have chosen considers parity as a helper function (i.e., neither a method nor an associated function to any of the existing traits); this was prompted by Pratyush's comment and this is my interpretation of it.

Reply:
@Pratyush do you prefer it to be an stand alone function or is it better if it is implemented as a function of SWMap struct (as in impl SWUMap)? In the second sence SWUMap has access to SWUParams which is already aware of the BaseField.

Reply:
Sure, making it a method of SWUMap also makes sense. I don’t have a strong opinion between free function and method